app/konnect/api/ksearch/query-syntax.md

```diff
@@ -0,0 +1,102 @@
+---
+title: KSearch
+content_type: reference
+---
+
+The {{site.konnect_short_name}} Search API [Beta] offers a powerful and flexible way to search through all {{site.konnect_short_name}} entities. It is designed to cater to a wide range of search requirements, enabling users to quickly find the information they need across different areas of the product.
+
+## Features
+
+**Global and Regional Access:** The KSearch API is available in global and regional locations with regional awareness. This ensures that entities returned are entities that are relevant to their geographical location, improving response times and conform to data residency expectations.
+
+**Comprehensive Response:** The responses of the search query are uniform and contain a fixed number of standard attributes: “id”, “type”, “labels/tags” and “name”. In addition, entity specific attributes are returned in a general attributes object.
+
+**Security and Accessibility** The KSearch API will only return entities that the user has permissions to access. If a user is able to retrieve the entity in the list endpoint, then the user will be able to see the entity in the search response.
+
+**Advanced Query Language**
+The KSearch API supports an advanced query language with the selectors, reserved characters and and logical operators.
+
+
+## Supported Entities
+- Users
+- Teams
+- System Accounts
+- Developers
+- Developer Teams
+- _More added soon._ (See `GET /search/type` for the latest entity types supported.)
+
+## Query Syntax
```

Comment:
@haydenlam For the query syntax (and the example queries you have), is that how you would search in the Konnect UI? Ex. I want to find the entity with "John" so I enter "name:John" into the Konnect UI search bar.

And to do the same with the API, would I do this?:
`curl -X 'GET' \
  'https://global.api.konghq.com/v0/search?q=%40name%3A%E2%80%9DJohn%22' \
  -H 'accept: application/json'`

Reply:
The search UI that is in design right now expects to allow a "basic" and an "advanced" input method where:

- "Basic" means the user will have guided suggestions and tags components that could be used to build a query.
- "Advanced" means exactly what you said, it's an exact representation of what the API accepts (except it does not have to be URL encoded)

Reply:
There's a sample tool posted here: https://kongstrong.slack.com/archives/C04UES9TGP7/p1702923542797189